webplugin/js/app/mck-sidebox-1.0.js

```diff
@@ -2961,7 +2971,13 @@ var userOverride = {
                     'aria-label',
                     LEAD_COLLECTION_LABEL.submit
                 );
-                leadCollectionHeading.innerHTML = LEAD_COLLECTION_LABEL.heading;
+                if (appOptions.preLeadCollection) {
+                    leadCollectionHeading.innerHTML =
+                        LEAD_COLLECTION_LABEL.heading;
+                } else {
+                    leadCollectionHeading.innerHTML =
+                        appOptions.appSettings.chatWidget.preChatGreetingMsg;
```

Comment:
can `chatWidget` or `preChatGreetingMsg` be `null`? if yes, should be add null checks here?

Reply:
made it `required `on the dashboard section if the user enables the pre-chat lead collection on dashboard settings to collect pre-chat lead

Reply:
what if someone introduced a bug in the system, and their code is deleting all the keys from `chatWidget` object, or even making it null? 
would it be better if we handle all unexpected things from here as well?